### ISSUE TITLE
File with G29 code causes nozzle to be pushed hard against the bed

### ISSUE THREAD

**Issue by michelluccote** [KevinOConnor/klipper#3536]
I calibrate my z-offset with klipper. I then run a file with a G28 and G29 command sequence. After auto bed leveling it tries to print and puts the nozzle right on the bed. To get it to print correctly I have to adjust the z-offset up by over 2mm before it starts to extrude filament. If I don't include the G29 it seems correctly calibrated. For now I have to do z-offset calibration. Run a G29 operation. Then start a print with-out a G29. Then its fine. 

My G29 macro is the following:
[gcode_macro G29]
default_parameter_T: 0
gcode:
  {% if printer.toolhead.status == "Printing" %}
    {printer.gcode.action_respond_info("This command cannot be used while printing")}
  {% elif printer.toolhead.homed_axes != "xyz" %}
    {printer.gcode.action_respond_info("Please home XYZ first")}
  {% else %}
    SAVE_GCODE_STATE NAME=G29_state
    G90
    G1 Z10 F240
    {% if T|int > 30 %}
      M190 S{T}
    {% endif %}
    BED_MESH_CALIBRATE
    {% if 'S' in params %}
      M140 S{S}
    {% endif %}
    G90
    G1 Z10 F240
    G1 X252 Y225 F6000
    BED_MESH_PROFILE SAVE=default
    RESTORE_GCODE_STATE NAME=G29_state
  {% endif %}

**Comment by klipper-gitissuebot** [comment]
Hi @michelluccote,

It did not look like there was a Klipper log file attached to this ticket. The log file has been engineered to answer common questions the Klipper developers have about the software and its environment (software version, hardware type, configuration, event timing, and hundreds of other questions).

Unfortunately, too many people have opened tickets without providing the log. That consumes developer time; time that would be better spent enhancing the software. If this ticket references an event that has occurred while running the software then the Klipper log must be attached to this ticket. Otherwise, this ticket will be automatically closed in a few days.

For information on obtaining the Klipper log file see: https://github.com/KevinOConnor/klipper/blob/master/docs/Contact.md

The log can still be attached to this ticket - just add a comment and attach the log to that comment.

Best regards,
~ Your friendly GitIssueBot

PS: I'm just an automated script, not a human being.

**Comment by michelluccote** [comment]
No problem. I have something printing now. Once it is done I'll do a clean restart and reproduce the issue and upload the log file.

**Comment by michelluccote** [comment]
I have found that adding one more G28 after the G29 resolve this issue completely. I have added the G28 to my G29 macro to do this by default. If you still need a log I can remove it and I'm sure I can reproduce the problem.

**Comment by klipper-gitissuebot** [comment]
It appears the Klipper log file was not provided.
I'm closing this now, sorry. Feel free to attach the log to this ticket sometime in the future, or create a new issue once you can provide the log file.
